### ISSUE TITLE
FR: AEM Assets hosted videos

### ISSUE THREAD

**Issue by ZoeBeykirch** [hlxsites/vg-roadchoice-com#237]
### Context/Description

As a Road Choice content editor, I want an embed video player solution so that I do not have to use YouTube in order to have a video hosted on my website. 

### Acceptance Criteria

**User AC:**
- [ ]  The video is able to play without having the need of social media cookies being set.

**Editor AC:**
- [ ] The content editor can copy and paste the generated code using the Sidekick AEM Assets tool
- [ ] The content editor can upload a custom thumbnail image for the standard video player. 

**Non-functional AC:**
- [ ] The video embed block should use Adobe Experience Manager Assets. 
- [ ] This ticket is to only deliver the standard video experience. 

### Out of scope

- Enhancements for autoplay, loop, hide controls, disable picture in picture, poster, aspect ratio, etc will be taken up in a separate ticket. 
- Anything that is not OOTB for AEM Asset videos, this should simply be reusing what has been done for Mack Trucks and Volvo Trucks in the Road Choice repository. 

**Related Tickets:** 
https://github.com/hlxsites/vg-volvotrucks-us/issues/610
https://github.com/hlxsites/vg-macktrucks-com/issues/649 
https://github.com/hlxsites/vg-macktrucks-com/issues/650

**Comment by girishdigrajkar** [comment]
@amglebihan: Validated the FR and below are the observation : 

**Across all viewports:**
1.  Verify CSS URLs error message is shown in console

**Desktop**:  All good; getting correct results.

**Mobile Viewport:**
    **iPhone:  Safari &  Chrome:**  
                  1. Video which is set to autoplay is not played automatically 
                  2. Video with the loop halts and never plays in loop
                  3.  Video with controls disabled shows full screen and Picture in Picture buttons and functions as well.
                  4. Video with controls disabled shows full screen and Picture in Picture buttons and functions as well.
                  5.  Poster is not displayed for video with poster
                  6.  4 videos on page does not show preview frame instead shown with black screen; upon playing the video is correct though.
     **Android: Chrome:** All good, getting correct results.
    

**Tablet Viewport:**
        **Safari &  Chrome:**  
        1. Video which is set to autoplay is not played automatically 
        2. Video with the loop halts and never plays in loop
        3.  Video with controls disabled shows full screen and Picture in Picture buttons and functions as well.
        4. Video with controls disabled shows full screen and Picture in Picture buttons and functions as well.
        5.  Poster is not displayed for video with poster
        6.  4 videos on page does not show preview frame instead shown with black screen; upon playing the video is correct though.

**Comment by amglebihan** [comment]
I contacted Ashish Chopra and Harsh Chiki from Adobe regarding the issues Girish mentioned above and am awaiting feedback. It appears that the video player doesn't initialize properly on iOS devices because a different code is being run for iPhone/iPad.

**Comment by ZoeBeykirch** [comment]
@girishdigrajkar @amglebihan Can you please share the link for testing?

**Comment by girishdigrajkar** [comment]
@ZoeBeykirch :  Tested with [this](https://237-aem-assets-hosted-videos--vg-roadchoice-com--hlxsites.aem.page/drafts/alan/v2-embed-multiple-config ) URL

**Comment by ZoeBeykirch** [comment]
@girishdigrajkar Thanks for sharing the link. Essentially, we are seeing the same issues that are captured in this ticket, right? https://github.com/hlxsites/vg-volvotrucks-us/issues/629

**Comment by ZoeBeykirch** [comment]
@girishdigrajkar Ok I have created a new ticket for addressing the issues you have noted, can you please review it and let me know if there's anything I left out or can remove from this ticket? https://github.com/hlxsites/vg-roadchoice-com/issues/260 Hopefully now we can unblock this current ticket which I have changed the requirements to only deliver the standard video player @amglebihan

**Comment by amglebihan** [comment]
@ZoeBeykirch I think that we can unblock the ticket. The standard video player works fine (if nothing else is spotted by @girishdigrajkar). We can tackle the issues with iOS in this other ticket you have created.

**Comment by girishdigrajkar** [comment]
Based on above comments since, iOS issues will be separately tackled hence, @amglebihan please proceed closing this ticket as results are as expected for android and desktop browsers:

Desktop:
<img width="1724" alt="image" src="https://github.com/user-attachments/assets/9b9adaa3-d459-4920-8f09-07e710b037dd">

<img width="1724" alt="image" src="https://github.com/user-attachments/assets/71ad061c-b3b9-45ca-abf4-d790206508d8">

Android phone:
<img width="1724" alt="image" src="https://github.com/user-attachments/assets/b35effac-e3fb-4e76-ace6-71558911d773">
